app/models/missing_link_finder.rb

```diff
@@ -10,8 +10,28 @@ def initialize(redirection)
   end
 
   def run
-    html = fetch(redirection.url)
+    begin
+      missing = missing_links
```

Comment:
It's kind of gross that `missing` can be `nil` or an empty array or a non-empty array.